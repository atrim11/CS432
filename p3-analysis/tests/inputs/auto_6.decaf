def int subtract(int x, int y) {
    return x - y;
}

def void main(int argc, int argv) {
    // Type mismatch in function call
    int result;
 result = subtract("text", 10);  // Error: first parameter should be int

    // Array bounds issu
    numbers[4] = 10;  // Error: out of bounds

    // Incorrect loop conditions
    while (result + 10) {  // Error: non-boolean used in while loop
        result = result - 1;
    }

    // Nested function calls with mismatches
    result = subtract(subtract(10, "string"), 5);  // Error: wrong type in nested call

    // Redeclaration of variables
    x = 5;
    x = 10;  // Error: redeclaration of 'x'

    // Undefined variables
    result = x + y;  // Error: 'y' is undeclared

    incorrectReturn();

    // Invalid boolean in if statement
    if (result - 10) {  // Error: non-boolean in if statement
        print_str("Invalid condition");
    }

    // Misuse of void function in expression
    result = printMax(10) + 5;  // Error: void function used in expression
}
