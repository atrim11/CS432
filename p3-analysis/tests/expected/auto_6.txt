Type mismatch in parameter 0 of call to 'subtract': expected int but found str on line 8
Symbol 'numbers' undefined on line 11
Type mismatch: bool expected but int found on line 14
Type mismatch in parameter 1 of call to 'subtract': expected int but found str on line 19
Symbol 'x' undefined on line 22
Symbol 'x' undefined on line 23
Symbol 'x' undefined on line 26
Symbol 'y' undefined on line 26
Symbol 'incorrectReturn' undefined on line 28
Type mismatch: bool expected but int found on line 31
Symbol 'printMax' undefined on line 36
'main' must return an integer
'main' must take no parameters
